nemo/collections/tts/models/univnet.py

```diff
@@ -0,0 +1,373 @@
+                            caption=f"gt mel {i}",
+                        ),
+                    ]
+
+            self.logger.experiment.log({"audio": clips, "specs": specs})
+
+    def _bias_denoise(self, audio, mel):
+        def stft(x):
+            comp = torch.stft(x.squeeze(1), n_fft=1024, hop_length=256, win_length=1024)
+            real, imag = comp[..., 0], comp[..., 1]
+            mags = torch.sqrt(real ** 2 + imag ** 2)
+            phase = torch.atan2(imag, real)
+            return mags, phase
+
+        def istft(mags, phase):
+            comp = torch.stack([mags * torch.cos(phase), mags * torch.sin(phase)], dim=-1)
+            x = torch.istft(comp, n_fft=1024, hop_length=256, win_length=1024)
+            return x
+
+        # create bias tensor
+        if self.stft_bias is None or self.stft_bias.shape[0] != audio.shape[0]:
+            audio_bias = self(spec=torch.zeros_like(mel, device=mel.device))
+            self.stft_bias, _ = stft(audio_bias)
+            self.stft_bias = self.stft_bias[:, :, 0][:, :, None]
+
+        audio_mags, audio_phase = stft(audio)
+        audio_mags = audio_mags - self.cfg.get("denoise_strength", 0.0025) * self.stft_bias
+        audio_mags = torch.clamp(audio_mags, 0.0)
+        audio_denoised = istft(audio_mags, audio_phase).unsqueeze(1)
+
+        return audio_denoised
+
+    def __setup_dataloader_from_config(self, cfg, shuffle_should_be: bool = True, name: str = "train"):
+        if "dataset" not in cfg or not isinstance(cfg.dataset, DictConfig):
+            raise ValueError(f"No dataset for {name}")
+        if "dataloader_params" not in cfg or not isinstance(cfg.dataloader_params, DictConfig):
+            raise ValueError(f"No dataloder_params for {name}")
+        if shuffle_should_be:
+            if 'shuffle' not in cfg.dataloader_params:
+                logging.warning(
+                    f"Shuffle should be set to True for {self}'s {name} dataloader but was not found in its "
+                    "config. Manually setting to True"
+                )
+                with open_dict(cfg["dataloader_params"]):
+                    cfg.dataloader_params.shuffle = True
+            elif not cfg.dataloader_params.shuffle:
+                logging.error(f"The {name} dataloader for {self} has shuffle set to False!!!")
+        elif not shuffle_should_be and cfg.dataloader_params.shuffle:
+            logging.error(f"The {name} dataloader for {self} has shuffle set to True!!!")
+
+        dataset = instantiate(cfg.dataset)
+        return torch.utils.data.DataLoader(dataset, collate_fn=dataset.collate_fn, **cfg.dataloader_params)
+
+    def setup_training_data(self, cfg):
+        self._train_dl = self.__setup_dataloader_from_config(cfg)
+
+    def setup_validation_data(self, cfg):
+        self._validation_dl = self.__setup_dataloader_from_config(cfg, shuffle_should_be=False, name="validation")
+
+    @classmethod
+    def list_available_models(cls) -> 'Optional[Dict[str, str]]':
```

Comment:
It is related to HiFiGAN. Need to update according to Univnet or remove currently.

Reply:
Removed it with [79d8a91](https://github.com/NVIDIA/NeMo/pull/3186/commits/79d8a9120c1c44f21bbcf144f612b588c7db98f8)